plugins/browser-plugin-media/src/api.ts

```diff
@@ -21,24 +21,30 @@ import {
   MediaTrackErrorArguments,
   MediaTrackSelfDescribingEventArguments,
   EventWithContext,
+  MediaPluginOptions,
 } from './types';
+import { setEventSpecificationContext } from './utils';
 
 export { MediaAdBreakType as MediaPlayerAdBreakType };
 
 const _trackers: Record<string, BrowserTracker> = {};
 const _context: Record<string, SelfDescribingJson[]> = {};
+const defaultPluginOptions = {};
 
 /**
  * Adds media tracking
  */
-export function SnowplowMediaPlugin(): BrowserPlugin {
+export function SnowplowMediaPlugin(pluginOptions: MediaPluginOptions = defaultPluginOptions): BrowserPlugin {
   let trackerId: string;
   return {
     activateBrowserPlugin: (tracker) => {
       trackerId = tracker.id;
       _trackers[trackerId] = tracker;
       _context[trackerId] = [];
     },
+    beforeTrack(payloadBuilder) {
```

Comment:
This is fine but seems a bit of a roundabout way for adding the information. All events are constructed in the `track()` function in this file – look for the `dispatchToTrackersInCollection()` calls at the end of this file (line 711). Wouldn't it be better to add the information there?